File: SuperShooter/Assets/Scripts/MoveOnCollide.js
var triggerTag : String;
var destination : GameObject;
var objectToRelocate : GameObject;


function OnTriggerEnter (c : Collider) : void{
		
		if (c.gameObject.tag == triggerTag){
			
			objectToRelocate.transform.position = destination.transform.position;
		}
    
}

File: SuperShooter/Assets/Scripts/MoveOnCollide.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class MoveOnCollide : MonoBehaviour
{
    public string triggerTag;
    public GameObject destination;
    public GameObject objectToRelocate;
    public virtual void OnTriggerEnter(Collider c)
    {
        if (c.gameObject.tag == this.triggerTag)
        {
            this.objectToRelocate.transform.position = this.destination.transform.position;
        }
    }

}